shell/platform/android/io/flutter/plugin/editing/InputConnectionAdaptor.java

```diff
@@ -128,36 +128,47 @@ public boolean setSelection(int start, int end) {
 
     @Override
     public boolean sendKeyEvent(KeyEvent event) {
-        final boolean result = super.sendKeyEvent(event);
-        if (event.getAction() == KeyEvent.ACTION_UP) {
+        if (event.getAction() == KeyEvent.ACTION_DOWN) {
             if (event.getKeyCode() == KeyEvent.KEYCODE_DEL) {
                 int selStart = Selection.getSelectionStart(mEditable);
                 int selEnd = Selection.getSelectionEnd(mEditable);
                 if (selEnd > selStart) {
                     // Delete the selection.
                     Selection.setSelection(mEditable, selStart);
-                    deleteSurroundingText(0, selEnd - selStart);
```

Comment:
Looks like you can delete `deleteSurroundingText` now. I don't see any other uses.

Reply:
Apologies - ignore that; it's part of the public interface; leave it for now and we can take a look at cleaning up separately.

Reply:
sounds good, anything else on the review?